src/celeritas/io/RootStepWriter.hh

```diff
@@ -25,20 +26,64 @@ namespace celeritas
  * entry a step. Since the ROOT data is stored in branches with primitive types
  * instead of a full struct, no dictionaries are needed for reading the output
  * file.
+ *
+ * The data can be filtered by defining a `RSWFilter`. This `std::function` can
+ * be user-defined and expanded as needed.
  */
 class RootStepWriter final : public StepInterface
 {
+  public:
+    // Unspecified opaque id
+    static constexpr size_type unspecified()
+    {
+        return static_cast<size_type>(-1);
+    }
+
+    // Unspecified array (dir and pos)
+    static constexpr std::array<real_type, 3> unspecified_array()
+    {
+        return std::array<real_type, 3>{0, 0, 0};
+    }
```

Comment:
These look like simply default values, as opposed to being tested elsewhere: is there a reason they're not just zero-initialized in the definitions below? More importantly, since `{0,0,0}` is a legitimate (and indeed common) point in space, it's not a good sentinel value.